src/test/java/com/reservation/studio/domain/StudioTest.java

```diff
@@ -0,0 +1,192 @@
+package com.reservation.studio.domain;
+
+import com.reservation.studio.domain.exception.ContactDetailsCannotBeBlankException;
+import com.reservation.studio.domain.exception.InvalidCancellationThresholdException;
+import com.reservation.studio.domain.exception.InvalidStudioNameException;
+import com.reservation.studio.domain.exception.StudioAlreadyActiveException;
+import com.reservation.studio.domain.exception.StudioAlreadyClosedException;
+import org.junit.jupiter.api.Test;
+
+import java.time.DayOfWeek;
+import java.time.Duration;
+import java.time.LocalTime;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class StudioTest {
+    @Test
+    void shouldCreateStudioAndEmitCreatedEvent() {
+        Studio studio = new StudioTestBuilder().build();
+
+        assertAll(
+                () -> assertFalse(studio.isActive()),
+                () -> assertThat(studio.pullEvents()).anyMatch(e -> e instanceof StudioCreated)
+        );
+    }
+
+    @Test
+    void shouldRenameStudioAndEmitRenamedEvent() {
+        Studio studio = new StudioTestBuilder().build();
+        StudioName newName = new StudioName("New Studio Name");
+
+        studio.rename(newName);
+
+        assertAll(
+                () -> assertTrue(studio.hasName(newName)),
+                () -> assertThat(studio.pullEvents()).anyMatch(e -> e instanceof StudioRenamed)
+        );
```

Comment:
Spoko, że mamy AssertJ, tylko trzeba się zdecydować na używanie jednej opcji w całym codebase, bo widzę, że niektóre testy korzystają tylko z JUnitowych asercji 😁

Nie mam też przekonania co do mieszania asercji z JUnit 5 i AssertJ w obrębie jednego testu.
Technicznie to co prawda nie przeszkadza, ale wydaje mi się, że to trochę "code smell" (chociażby w [tym](https://wearecommunity.io/communities/testautomation/articles/6877) artykule w sekcji Best practices piszą, żeby tak nie robić).

Z wykorzystaniem samego assertJ można to zrobić np. tak:
```suggestion
        assertSoftly(softly -> {
            softly.assertThat(studio.hasName(newName)).isTrue();
            softly.assertThat(studio.pullEvents()).anyMatch(StudioRenamed.class::isInstance);
        });
```